INTRODUCTION.md

```diff
@@ -388,6 +392,244 @@ may be produced out of order, since a sub-sequent message in a sub-sequent
 ProduceRequest may already be in-flight (and accepted by the broker)
 by the time the retry for the failing message is sent.
 
+Using the Idempotent Producer prevents reordering even with `max.in.flight` > 1,
+see [Idempotent Producer](#idempotent-producer) below for more information.
+
+
+### Idempotent Producer
+
+librdkafka supports the idempotent producer which provides strict ordering and
+and exactly-once producer guarantees.
+The idempotent producer is enabled by setting the `enable.idempotence`
+configuration property to `true`, this will automatically adjust a number of
+other configuration properties to adhere to the idempotency requirements,
+see the documentation of `enable.idempotence` in [CONFIGURATION.md] for
+more information.
+
+
+#### Guarantees
+
+There are three types of guarantees that the idempotent producer can satisfy:
+
+ * Exactly-once - a message is only written to the log once.
+                  Does NOT cover the exactly-once consumer case.
+ * Ordering - a series of messages are written to the log in the
```

Comment:
would be good to clarify whether this does or does not also imply exactly once. The way i read it is that it does not, but i'm left with some confusion.

Reply:
This lists what guarantees the idemp producer satisfies; not necessarily which guarantees you can select to use.
Ordering and exactly-once are always enabled with idemp, gapless was added to cover corner cases which the Java producer handles in a different way.

I'll clarify